jerry-core/debugger/jerry-debugger-ws.c

```diff
@@ -13,8 +13,6 @@
  * limitations under the License.
  */
 
-#ifdef JERRY_DEBUGGER
```

Comment:
The reason is "transitive". CMakeLists added all sources of jerry-core to `SOURCE_CORE_FILES` unconditionally but the sources of the debugger. And the reason for that, I guess, was that the sources were completely guarded by `JERRY_DEBUGGER`. However, if the guard was off, the compilation unit was empty, which was a warning/error for C99. I wanted to treat all sources equal: it's "nicer" but also easier to maintain. The common workaround for the empty compilation unit problem is to have at least a typedef somewhere, which is usually ensured by having the includes outside the guard. The jerry-core/ecma subdir uses the same approach at various places, too, e.g., ecma-builtin-promise.c. (We don't add those sources to the list of files conditionally either.)

The strange thing is that Travis CI tests jerry-libc builds and it does _not_ fail... (Although I tend to agree with you that some of the headers here should not be available outside the guard.)